### DISCUSSION TITLE
Contribute to Verified Tuya v2 Product List

### DISCUSSION THREAD

**Discussion by zlinoliver** [tuya/tuya-home-assistant#309] [General]
This product list includes devices submitted by the community and confirmed working with Tuya v2 Home Assistant integration.

## Supported Device Categories

- **Climate**: Supports air conditioner, heater, thermostat.
- **Cover**: Supports curtain, curtain switch.
- **Fan**: Supports fan, air purifier.
- **Light**: Supports light, light strip, ambient light, light string, humidifier's light.
- **Scene**: Supports smart scene.
- **Switch**: Supports switch, socket, power strip, smart kettle, breaker, pet water feeder, air purifier, diffuser.

<table class="tableizer-table">

<thead><tr class="tableizer-firstrow"><th>First Class Category</th><th>Category code</th><th>Category[en]</th><th>Support Version</th></tr></thead><tbody>
 <tr><td>Large Home Appliances</td><td>kt</td><td>Air conditioner</td><td>1.1</td></tr>
 <tr><td>Small Home Appliances</td><td>cwysj</td><td>Pet Water Feeder </td><td>1.3</td></tr>
 <tr><td>&nbsp;</td><td>qn</td><td>Heater</td><td>1.1</td></tr>
 <tr><td>&nbsp;</td><td>sd</td><td>Robot Vaccum</td><td>1.3</td></tr>
 <tr><td>&nbsp;</td><td>jsq</td><td>Humidifier</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>fs</td><td>Fan</td><td>1.1</td></tr>
 <tr><td>&nbsp;</td><td>cl</td><td>Curtain</td><td>1.1</td></tr>
 <tr><td>&nbsp;</td><td>kj</td><td>Air Purifer</td><td>1.3.2</td></tr>
 <tr><td>Electrician Category</td><td>kg</td><td>Switch</td><td>1.0</td></tr>
 <tr><td>&nbsp;</td><td>cz</td><td>Socket</td><td>1.0</td></tr>
 <tr><td>&nbsp;</td><td>pc</td><td>Power Strip</td><td>1.1</td></tr>
 <tr><td>&nbsp;</td><td>clkg</td><td>Curtain Switch</td><td>1.1</td></tr>
 <tr><td>&nbsp;</td><td>wk</td><td>Thermostat</td><td>1.3</td></tr>
 <tr><td>&nbsp;</td><td>dlq</td><td>Circuit Breaker</td><td>1.3.1</td></tr>
 <tr><td>Security & Sensors</td><td>wsdcg</td><td>Temperature and Humidity Sensor</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>mcs</td><td>Door Window Sensor</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>ywbj</td><td>Smoke Detector</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>rqbj</td><td>Gas Detector</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>pir</td><td>PIR Detector</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>sj</td><td>Water Detector</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>pm2.5</td><td>PM2.5 Sensor</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>hps</td><td>Human Presence Sensor</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>sos</td><td>Emergency Button</td><td>1.2</td></tr>
 <tr><td>&nbsp;</td><td>ldcg</td><td>Luminance Sensor</td><td>1.3.1</td></tr>
 <tr><td>&nbsp;</td><td>ms</td><td>Lock</td><td>1.3.2</td></tr>
 <tr><td>Lighting</td><td>dj</td><td>Light</td><td>1.0</td></tr>
 <tr><td>&nbsp;</td><td>dd</td><td>Light Strip</td><td>1.1</td></tr>
 <tr><td>&nbsp;</td><td>fwl</td><td>Ambient light</td><td>1.1</td></tr>
 <tr><td>&nbsp;</td><td>dc</td><td>Light string</td><td>1.1</td></tr>
 <tr><td>&nbsp;</td><td>xdd</td><td>Ceiling Light</td><td>1.4.0</td></tr>
 <tr><td>Kitchen Appliances</td><td>bh</td><td>Smart Kettle</td><td>1.2</td></tr>
</tbody></table>

## Verified PBT Products Template

If you have tested and confirmed any Tuya v2 compatible PBT device, please leave a comment below in the following format to share with the community. We appreciate your contribution!

- Brand: AcooSun
- Model: E27
- Category: Light
- Category Code: dj
- Suggested Purchase link: <https://www.aliexpress.com/item/1005002657974573.html>

**Comment by lashlosh** [comment]
RGB Light led strip, after integration with home assistant it shows the colors, it lights by default in white, but when I select any color the color changed only on the application but the strip doesn't change

**Comment by MarkWattTech** [comment]
Brand | Model | Category | Category Code| Suggested Purchase Link
------------ | ------------- | -------------  | ------------- | -------------  |
Novostella | B22 | Light | dj | https://www.amazon.co.uk/gp/product/B07DN4NLKM/

**Comment by AngloYorkshire** [comment]
It seems that the Tesco Calex products are working but except for this problem
Added a Calex Smart Plug onto Lovelace as a Clickable Graphic Switch but it only works one way either on or off and you cannot switch it the other way after.
